tests/src/Kernel/ConfigEntityTest.php

```diff
@@ -66,6 +76,46 @@ public function testConfigAction() {
   }
 
   /**
+   * Test saving a deeply nested rule.
+   *
+   * @todo This test currently fails if $strictConfigSchema = TRUE.
+   */
+  public function testNestedRuleSave() {
+    // This test is specific to reaction rules, so get the appropriate
+    // entity storage:
+    $storage = $this->container->get('entity.manager')->getStorage('rules_reaction_rule');
+    // Create a nested expression.
+    $and = $this->expressionManager->createAnd()
+      ->addCondition('rules_test_false')
+      ->addCondition('rules_test_true', ContextConfig::create()->negateResult())
+      ->negate();
+
+    $or = $this->expressionManager->createOr()
+      ->addCondition('rules_test_true', ContextConfig::create()->negateResult())
+      ->addCondition('rules_test_false')
+      ->addCondition($and);
+
+    $rule = $this->expressionManager->createRule();
```

Comment:
We can only change our own config entity implementation.  We need to remove anything other than arrays and primitives before we call the parent:: methods, or we get the original crash.

I can certainly move the code out of the expression hierarchy into the config entity class, but IIRC it's a bit easier if I can see protected data; the methods are simpler and easier to understand.  I'll take another look at this to see if this is